CONTRIBUTING.md

```diff
@@ -189,6 +190,16 @@ Trestle relies on reference data from two NIST repositories for testing:
 
 Both of these repositories are submodules in the trestle project. In order to develop / test trestle the submodules must be checked out with `git submodule update --init` or `make submodules`.
 
+### Code testing
```

Comment:
Is it needed by the requirement to add here that is mandatory to execute all the current (and new tests) to make CI pass correctly (I mean executing make test locally) prior merging to main? For me it is kind of implicit otherwise you're not going to pass the checks, but just asking. Or is it just mentioned in the requirement to document the test coverage? I don't seem to see any specification anywhere on the [best practices criteria](https://www.bestpractices.dev/en/criteria/0) page

Reply:
Here is the link https://www.bestpractices.dev/en/projects/9408/edit#test_policy:~:text=It%20is%20SUGGESTED%20that%20this%20policy%20on%20adding%20tests%20(see%20test_policy)%20be%20documented%20in%20the%20instructions%20for%20change%20proposals.

The text is also copied into the corresponding issue #1696. 

This OpenSSF best practice is a SUGGESTION. My thought is to add some appropriate text (thus this PR) and check the Met box. Feel free to suggest alternative text.

Reply:
@degenaro I'm all for stating this, however, this is not the only gate we are measuring.

pytest will fail if the total code covered is < 96% on the trestle directory. 

In addition failure occurs if any of the following conditions occur: 

![image](https://github.com/user-attachments/assets/81bd3cd4-da79-44f9-b857-b0cbf211ac53)

The gate is defined here: https://sonarcloud.io/organizations/compliance-trestle/quality_gates/show/50761 which is accessible for the public.

I think we should 
1. Highlight those conditions
2. Align the total coverage mechanism in sonar with pytest OR remove one of the metrics.